src/Mod/TechDraw/Gui/PagePrinter.cpp

```diff
@@ -423,3 +423,25 @@ PaperAttributes::PaperAttributes()
     pagewidth = A4Heightmm;
     pageheight = A4Widthmm;
 }
+
+void PagePrinter::resetPageState(ViewProviderPage* vpPage, bool savedState)
+{
+    auto pageFeature  = vpPage->getDrawPage();
+    if (!pageFeature) {
+        // how did this happen?
+        // if we have a view provider for a page, but no page, something hase gone badly
+        // wrong somewhere
+        return;
+    }
+
+    auto guiDoc = vpPage->getDocument();
+    if (!guiDoc) {
+        // same this here.  can't really have a view provider without a document
+        return;
+    }
+    auto scene = vpPage->getQGSPage();
+    scene->setExportingPdf(false);
+    scene->setExportingSvg(false);
+    guiDoc->setModified(savedState);
+    pageFeature->redrawCommand();
+}
```

Comment:
This is the exact copy&paste of `MDIViewPage::resetPageExportState`, see review comments in #17488 for more.

Reply:
Both `setExportingPdf` and `setExportingSvg` in `PagePrinter::resetPageState` and `MDIViewPage::resetPageExportState` depends on  `getDrawPage()`. Why? `PagePrinter::printAllPdf` does not seem to care about it. Also it does not seem we should skip that in case there is no way to call `redrawCommand()`.
With flaws fixed, those functions could live in `DrawGuiUtil.cpp`